develop/tutorials/articles/02-from-liferay-6-to-liferay-7/07-upgrading-plugins-to-liferay-7/07-upgrading-application-upgrade-and-verifier-processes.markdown

```diff
@@ -0,0 +1,385 @@
+        )
+
+        public class CalendarServiceUpgrade implements UpgradeStepRegistrator       
+
+4.  Remove the intermediate classes that wrapped the internal steps, i.e the 
+    `public class UpgradeProcess_1_0_1 extends UpgradeProcess...`.
+    
+    Following the prior upgrade process in 6.2, you had to define the property 
+    `upgrade.processes`, a list of `UpgradeProcesses` representing the different 
+    upgrades for a specific version of your module, in your 
+    `portal-ext.properties`.
+    
+    For instance, the code below shows the previous process for upgrading 
+    Calendar-service module from v1.0.0 to 1.0.1 and then to 1.0.2. 
+    
+        upgrade.processes=
+            com.liferay.calendar.hook.upgrade.UpgradeProcess_1_0_0,
+            com.liferay.calendar.hook.upgrade.UpgradeProcess_1_0_1,
+            com.liferay.calendar.hook.upgrade.UpgradeProcess_1_0_2
+    
+    Each step between versions was represented by a single class extending
+    `UpgradeProcess`, using a method called `doUpgrade`. This method was 
+    responsible for executing the internal steps to update the database to that 
+    concrete version. A method `getThreshold` is provided also to specify the 
+    schema version where the upgrade starts.
+    
+    Whenever you needed another internal step, you added another
+    `upgrade(new UpgradePortletPreferences());` etc. after the existing ones.
+
+    With the new framework, previous type of classes are represented by upgrade 
+    registrations instead.
+
+    Each upgrade is represented by an upgrade registration. An upgrade 
+    registration is an abstraction for the changes you need to apply to the 
+    database from one version to the next one.
+
+    To define a registration, you need to provide
+
+    - the bundle symbolic name of the module.
+    - the schema version your module wants to upgrade from (as a `String`)
+    - the schema version your module wants to upgrade to (as a `String`)
+    - a list of `UpgradeSteps` instances
+
+    For example, here is an upgrade process for the 
+    `com.liferay.calendar.service` module:
+
+        @Override
+    
+        public void register(Registry registry) {
+        
+            registry.register(
+                "com.liferay.calendar.service", "0.0.1", "1.0.0",
+                new com.liferay.calendar.upgrade.
+                v1_0_0.UpgradeCalendarBooking()));    
+        }
+
+    In this example, the `com.liferay.calendar.service` module is being upgraded 
+    from version 0.0.1 to version 1.0.0. The changes are produced by a list of 
+    `UpgradeSteps` instances, which in this example contains only one step:
+
+        new com.liferay.calendar.upgrade.v1_0_0.UpgradeCalendarBooking()
+
+    The internal steps defined within the intermediate classes, the former 
+    `UpgradeProcess` class, as they are indeed `UpgradeSteps`, require no change
+    on your part. The new framework will process the steps as they are. The
```

Comment:
I'm not sure what the first sentence "The internal steps..." means. As I read it, here's how I understand it ...
- The internal steps defined the former `UpgradeProcess` class within the intermediate classes.
- The internal steps are `UpgradeSteps` instances.
- You don't need to change the internal steps.

Can you clarify? Thanks.